### PR TITLE
8.0 manage invalid data error

### PR DESCRIPTION
**Pull request by gfcapalbo**
Problem:
I would have many jobs stuck in start.  THese jobs Had all raised InvalidDataError.
Strangely the Exception was not catched in main.py, but the generic 

Therefore the store function that had to write my failed job would never trigger

When explicitly importing InvalidDataError, the exception was catched (see commit  
https://github.com/gfcapalbo/connector/commit/5b77f3c6d77a9e10ee3f0f2e2fca72747e0f0244)

I replaced except (FailedJobError, InvaliDataError, Exception):  
with 
except:

I am still not sure why this works for me, but it seems OK to set ALL of the forseeable exceptions to fail jobs.

### CODE REVIEW

**Comment by gfcapalbo** [comment]
@StefanRijnhart   @hbrunn     what do you think?

**Comment by hbrunn** [comment]
I think first of all, you need to propose to the correct target branch, which is 8.0. Then I think @sbidoul and @guewen are a lot better persons to ask for an opinion because to my knowledge, this repo is their baby

File: connector/controllers/main.py [hunk 1]
```diff
@@ -112,11 +112,10 @@ def retry_postpone(job, message, seconds=None):
             retry_postpone(job, unicode(err), seconds=err.seconds)
             _logger.debug('%s postponed', job)
 
-        except (FailedJobError, Exception):
+        except:
```
Comment (sbidoul) [inline connector/controllers/main.py:115]: AFAICS, `InvalidDataError` is a `ConnectorException`, which is `RuntimeError` which is an `Exception`, so changing this exception handler should not have any influence on your use case. Your root cause must be something else.

Comment (hbrunn) [inline connector/controllers/main.py:115]: @sbidoul that's what I thought too. But the same applies to `FailedJobError`, and this one is mentioned explicitly here. `git blame` says you wrote this line, do you remember why you added `FailedJobError`? And do you agree in general to catch unconditionally here? I'd think we want to persist the failed state no matter what happened, right?

Comment (sbidoul) [inline connector/controllers/main.py:115]: I was also wondering why the `FailedJobError` is there. This code comes from https://github.com/OCA/connector/blob/1dbde5d23dbd97c5fee4accd408cb619bfd36ce8/connector/queue/worker.py#L150 (@guewen's). 

Still, the current except statement is equivalent to `except Exception` which catches everything for all practical purposes (https://docs.python.org/2/library/exceptions.html#exceptions.Exception).

@gfcapalbo can you say more about your problem?

**Comment by gfcapalbo** [comment]
Closed/
Please continue discussion here : Correct PR https://github.com/OCA/connector/pull/188

**Pull request closed without merging**